src/NexusMods.App.UI/Pages/Downloads/InProgressView.axaml.cs

```diff
@@ -40,18 +42,34 @@ public InProgressView()
 
             this.BindCommand(ViewModel, vm => vm.ResumeAllTasksCommand, view => view.ResumeAllButton)
                 .DisposeWith(d);
-
-            this.BindCommand(ViewModel, vm => vm.ShowSettings, view => view.SettingsButton)
+            
+            this.BindCommand(ViewModel, vm => vm.HideSelectedCommand, view => view.ClearButton)
+                .DisposeWith(d);
+            
+            this.BindCommand(ViewModel, vm => vm.HideAllCommand, view => view.ClearAllButton)
                 .DisposeWith(d);
 
-            this.OneWayBind(ViewModel, vm => vm.Tasks, 
-                    view => view.ModsDataGrid.ItemsSource)
+            this.OneWayBind(ViewModel, vm => vm.InProgressTasks, 
+                    view => view.InprogressDataGrid.ItemsSource)
+                .DisposeWith(d);
+            
+            this.OneWayBind(ViewModel, vm => vm.CompletedTasks, 
+                    view => view.CompletedDataGrid.ItemsSource)
                 .DisposeWith(d);
 
             this.WhenAnyValue(view => view.ViewModel!.Columns)
-                .GenerateColumns(ModsDataGrid)
+                .GenerateColumns(InprogressDataGrid)
                 .DisposeWith(d);
 
+            // Fix the CompletedDataGrid Width when number of items changes
+            this.WhenAnyValue(view => view.ViewModel!.CompletedDownloadCount)
+                .OnUI()
```

Comment:
`OnUI` statements should be in the VM, the View shouldn't have to do thread switching.